backend/app/services/adapters/task_kinds/queries.py

```
@@ -37,26 +37,140 @@
+        filtered_task_ids = []
+        for row in all_tasks_result:
+            task_id, task_json = row
+            try:
+                task_crd = Task.model_validate(task_json)
+            except Exception:
+                continue
+
+            status = task_crd.status.status if task_crd.status else "PENDING"
+            if status == "DELETE":
+                continue
+            if is_background_task(task_crd):
```

Ruff (0.15.5)

[error] Undefined name `is_background_task`

(F821)